From cdb388bee6c34a72e3481669adc8f21d978aa1e2 Mon Sep 17 00:00:00 2001
From: Gustavo Solaira <gustavos@codeaurora.org>
Date: Wed, 14 Dec 2016 14:09:40 -0800
Subject: [PATCH] build: Stub out clang and java packages

Force LOCAL_CLANG to be false because clang is not used
in LE, also stub out all java packages makefiles since
it will not be used in LE either.

Change-Id: Ide9366cd1d1db09f3c461ac001f3646fd6db6eee
---
 build/core/binary.mk |  3 +++
 build/core/config.mk | 10 +++++-----
 build/core/stub.mk   |  0
 3 files changed, 8 insertions(+), 5 deletions(-)
 create mode 100644 build/core/stub.mk

diff --git a/build/core/binary.mk b/build/core/binary.mk
index 9178cdb..a748e78 100644
--- a/build/core/binary.mk
+++ b/build/core/binary.mk
@@ -141,6 +141,9 @@ ifdef LOCAL_CLANG_$($(my_prefix)$(LOCAL_2ND_ARCH_VAR_PREFIX)ARCH)
 my_clang := $(LOCAL_CLANG_$($(my_prefix)$(LOCAL_2ND_ARCH_VAR_PREFIX)ARCH))
 endif
 
+# Override clang for Android Compat
+my_clang := false
+
 # arch-specific static libraries go first so that generic ones can depend on them
 my_static_libraries := $(LOCAL_STATIC_LIBRARIES_$($(my_prefix)$(LOCAL_2ND_ARCH_VAR_PREFIX)ARCH)) $(LOCAL_STATIC_LIBRARIES_$(my_32_64_bit_suffix)) $(my_static_libraries)
 my_whole_static_libraries := $(LOCAL_WHOLE_STATIC_LIBRARIES_$($(my_prefix)$(LOCAL_2ND_ARCH_VAR_PREFIX)ARCH)) $(LOCAL_WHOLE_STATIC_LIBRARIES_$(my_32_64_bit_suffix)) $(my_whole_static_libraries)
diff --git a/build/core/config.mk b/build/core/config.mk
index f7a2f76..a335d30 100644
--- a/build/core/config.mk
+++ b/build/core/config.mk
@@ -79,9 +79,9 @@ BUILD_PHONY_PACKAGE:= $(BUILD_SYSTEM)/phony_package.mk
 BUILD_HOST_PREBUILT:= $(BUILD_SYSTEM)/host_prebuilt.mk
 BUILD_PREBUILT:= $(BUILD_SYSTEM)/prebuilt.mk
 BUILD_MULTI_PREBUILT:= $(BUILD_SYSTEM)/multi_prebuilt.mk
-BUILD_JAVA_LIBRARY:= $(BUILD_SYSTEM)/java_library.mk
-BUILD_STATIC_JAVA_LIBRARY:= $(BUILD_SYSTEM)/static_java_library.mk
-BUILD_HOST_JAVA_LIBRARY:= $(BUILD_SYSTEM)/host_java_library.mk
+BUILD_JAVA_LIBRARY:= $(BUILD_SYSTEM)/stub.mk
+BUILD_STATIC_JAVA_LIBRARY:= $(BUILD_SYSTEM)/stub.mk
+BUILD_HOST_JAVA_LIBRARY:= $(BUILD_SYSTEM)/stub.mk
 BUILD_DROIDDOC:= $(BUILD_SYSTEM)/droiddoc.mk
 BUILD_COPY_HEADERS := $(BUILD_SYSTEM)/copy_headers.mk
 BUILD_NATIVE_TEST := $(BUILD_SYSTEM)/native_test.mk
@@ -93,8 +93,8 @@ BUILD_STATIC_TEST_LIBRARY := $(BUILD_SYSTEM)/static_test_lib.mk
 BUILD_HOST_STATIC_TEST_LIBRARY := $(BUILD_SYSTEM)/host_static_test_lib.mk
 
 BUILD_NOTICE_FILE := $(BUILD_SYSTEM)/notice_files.mk
-BUILD_HOST_DALVIK_JAVA_LIBRARY := $(BUILD_SYSTEM)/host_dalvik_java_library.mk
-BUILD_HOST_DALVIK_STATIC_JAVA_LIBRARY := $(BUILD_SYSTEM)/host_dalvik_static_java_library.mk
+BUILD_HOST_DALVIK_JAVA_LIBRARY := $(BUILD_SYSTEM)/stub.mk
+BUILD_HOST_DALVIK_STATIC_JAVA_LIBRARY := $(BUILD_SYSTEM)/stub.mk
 
 
 -include cts/build/config.mk
diff --git a/build/core/stub.mk b/build/core/stub.mk
new file mode 100644
index 0000000..e69de29
-- 
1.9.1

